lms/djangoapps/courseware/model_data.py

```diff
@@ -994,15 +994,25 @@ def set_score(user_id, usage_key, score, max_score):
     """
     Set the score and max_score for the specified user and xblock usage.
     """
-    student_module, created = StudentModule.objects.get_or_create(
-        student_id=user_id,
-        module_state_key=usage_key,
-        course_id=usage_key.course_key,
-        defaults={
-            'grade': score,
-            'max_grade': max_score,
-        }
-    )
+    created = False
+    kwargs = {"student_id": user_id, "module_state_key": usage_key, "course_id": usage_key.course_key}
+    try:
+        with transaction.atomic():
+            student_module, created = StudentModule.objects.get_or_create(
+                defaults={
+                    'grade': score,
+                    'max_grade': max_score,
+                },
+                **kwargs
+            )
+    except IntegrityError:
+        # log information for duplicate entry and get the record as above command failed.
+        student_module = StudentModule.objects.get(**kwargs)
+        log.warning("set_score: IntegrityError for student %d - course_id %s - usage_key %s having "
```

Comment:
Have you checked what is the current logging level in prod?  Are you expecting `log.warning`  to log in splunk?

Reply:
Here is how you would break the long logging lines.
https://openedx.atlassian.net/wiki/spaces/EDUCATOR/pages/251528559/Coding+Practices

Reply:
Yes, I have added `log.warning` to log in Splunk.

Reply:
Have you checked what is the current logging level in prod? I am not sure warnings are logged in splunk.

Reply:
@awaisdar001 I don't have any idea to check the current logging level in Prod. Can you tell me please?

Reply:
I have confirmed from Splunk. WARNINGS are also logged in Splunk.